Unable to install reason on nixos because there is no /bin/bash

Hi! On my system (nixos), `bash` does not live at `/bin/bash`, and it causes this error during installation:

```
-> % opam pin add -y reason 'https://github.com/facebook/reason.git#0.0.6'                                                                                           ~
[NOTE] Package reason is already git-pinned to https://github.com/facebook/reason.git#0.0.6.
       This will erase any previous custom definition.
Proceed ? [Y/n] y

[reason] https://github.com/facebook/reason.git#0.0.6 updated
[reason] Installing new package description from https://github.com/facebook/reason.git#0.0.6

reason needs to be installed.
The following actions will be performed:
  ∗  install reason 0.0.5*

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[reason] https://github.com/facebook/reason.git#0.0.6 already up-to-date

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of reason failed at "make compile_error".

#=== ERROR while installing reason.0.0.5 ======================================#
# opam-version 1.2.2
# os           unix
# command      make compile_error
# path         /home/matklad/.opam/4.02.3/build/reason.0.0.5
# compiler     4.02.3
# exit-code    2
# env-file     /home/matklad/.opam/4.02.3/build/reason.0.0.5/reason-1514-6d07ae.env
# stdout-file  /home/matklad/.opam/4.02.3/build/reason.0.0.5/reason-1514-6d07ae.out
# stderr-file  /home/matklad/.opam/4.02.3/build/reason.0.0.5/reason-1514-6d07ae.err
### stdout ###
# Makefile:40: recipe for target 'update_error' failed
### stderr ###
# make: /bin/bash: Command not found
# make: *** [update_error] Error 127



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions failed
  ∗  install reason 0.0.5
No changes have been performed
[NOTE] Pinning command successful, but your installed packages may be out of sync.
```

Looks like the culprit is [this line](https://github.com/facebook/reason/blob/27955a4669fef1f39b638165e11f1a7128beb236/Makefile#L3). Quick search through the repo reveals [one more](https://github.com/facebook/reason/blob/967e715960c550cc5b055e4222ae45b244f098a0/src/refmt_merlin_impl.sh#L1) hardcoded bash.

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 # Portions Copyright (c) 2015-present, Facebook, Inc. All rights reserved.
 
-SHELL=/bin/bash -o pipefail
+SHELL=bash -o pipefail
 
 default: build test
 

--- a/src/refmt_merlin_impl.sh
+++ b/src/refmt_merlin_impl.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 

